addons/mail/static/src/utils/common/hooks.js

```diff
@@ -388,21 +388,16 @@ export function useScrollPosition(refName, model, clearOn) {
     return self;
 }
 
-/**
- * @typedef {Object} MessageEdition
- * @property {composerOfThread} composerOfThread
- * @property {editingMessage} editingMessage
- * @property {function} exitEditMode
- * @returns {MessageEdition}
- */
 export function useMessageEdition() {
     const state = useState({
+        /** @type {import('@mail/core/common/composer').Composer} */
         composerOfThread: null,
+        /** @type {import('@mail/core/common/message_model').Message} */
```

Comment:
Note that it's best to type with `import("models").Composer`, the path of original class does not have patches, whereas `import("models")` does take patches into account